src/pages/ProductPage/ProductPageItem.jsx

```diff
@@ -0,0 +1,220 @@
+import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
+import FavoriteIcon from '@mui/icons-material/Favorite';
+import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
+import StarIcon from '@mui/icons-material/Star';
+import { useCart, useAuth, useWishList, useProduct } from 'contexts';
+import { postToCart, postToWishList } from 'utils/';
+import { useToast } from 'custom-hooks/useToast'
+import { useState, useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { deleteProductInWishList } from 'utils';
+
+const ProductPageItem = () => {
+
+    const { productId } = useParams();
+    const { products } = useProduct();
+    const { cartState: { cartItems }, cartDispatch } = useCart();
+    const { wishListState: { wishListItems }, wishListDispatch } = useWishList();
+    const { authState: { token, isAuth } } = useAuth();
+    const { showToast } = useToast();
+    const [ bookInCart, setBookInCart ] = useState(false);
+    const [ bookInWishList, setBookInWishList ] = useState(false);
+    const navigate = useNavigate();
+    const [isOngoingNetworkCall, setIsOngoingNetworkCall] = useState(false);
+
+    const book = products.find(product => product._id === productId);
+
+    const {
+        author,
+        bookType,
+        coverImg,
+        discountPercent,
+        genres,
+        _id,
+        offers,
+        originalPrice,
+        sellingPrice,
+        title,
+        totalRatings,
+        totalStars,
+    } = book;
+
+    useEffect(() => {
+        const isBookInCart = cartItems.find(cartItem => cartItem._id === _id)
+        const isBookInWishList = wishListItems.find(wishListItem => wishListItem._id === _id)
+        if(isBookInCart) {
+            setBookInCart(true);
+        }
+        else {
+            setBookInCart(false);
+        }
+
+        if(isBookInWishList) {
+            setBookInWishList(true);
+        }
+        else {
+            setBookInWishList(false);
+        }
+    }, []);
+
+    const outOfStock = !offers.inStock;
+
+
+    const localeOriginalPrice = originalPrice.toLocaleString("en-IN", {minimumFractionDigits: 2, maximumFractionDigits: 2});
+    const localeSellingPrice = sellingPrice.toLocaleString("en-IN", {minimumFractionDigits: 2, maximumFractionDigits: 2});
```

Comment:
I don't think you need to specify minimumFractionDigits: 2, I think it is by default set to 2.